base/src/main/resources/db/migration/V80__new_folder_structure.sql

```diff
@@ -0,0 +1,288 @@
+
+create table ehr.ehr_folder_history
+(
+    id                uuid,
+    ehr_id            uuid        NOT NULL,
+    ehr_folders_idx   int              NOT NULL,
+    row_num           int              NOT NULL,
+    contribution_id   uuid        NOT NULL,
+    audit_id          uuid        NOT NULL,
+    archetype_node_id TEXT,
+    path              TEXT[],
+    hierarchy_idx     text collate "C" not null,
+    hierarchy_idx_cap text collate "C" not null,
+    hierarchy_idx_len int              not null,
+    items          uuid[],
+    fields            jsonb,
+    namespace         TEXT default '1f332a66-0e57-11ed-861d-0242ac120002',
+    sys_version       INT         NOT NULL,
+    sys_period_lower  timestamptz NOT NULL,
+    sys_period_upper  timestamptz,
+    sys_deleted       boolean     NOT NULL,
+    PRIMARY KEY (ehr_id, id, sys_version),
+    FOREIGN KEY (ehr_id) REFERENCES ehr.ehr (id),
+    FOREIGN KEY (contribution_id) REFERENCES ehr.contribution (id),
+    FOREIGN KEY (audit_id) REFERENCES ehr.audit_details (id)
+);
+
+ALTER TABLE ehr.ehr_folder_history
+    ENABLE ROW LEVEL SECURITY;
+CREATE POLICY ehr_policy_all ON ehr.ehr_folder_history FOR ALL USING (namespace = current_setting('ehrbase.current_tenant')) WITH CHECK (namespace = current_setting('ehrbase.current_tenant'));
+
+-- Remove old folder structure
+
+alter table ehr.ehr
+    drop column if exists directory;
+drop table ehr.folder, ehr.folder_hierarchy, ehr.folder_items,ehr.folder_history,ehr.folder_items_history,ehr.folder_hierarchy_history,ehr.object_ref, ehr.object_ref_history;
+
+drop function ehr.admin_delete_folder(folder_id_input uuid);
+drop function ehr.admin_delete_folder_history(folder_id_input uuid);
+drop function ehr.admin_delete_folder_obj_ref_history(contribution_id_input uuid);
+
+create or replace function admin_delete_ehr_full(ehr_id_param uuid)
+    returns TABLE
+            (
+                deleted boolean
+            )
+    language plpgsql
+    security definer
+    SET search_path = ehr, pg_temp
+as
+$$
+BEGIN
+    -- Disable versioning triggers
+    ALTER TABLE ehr.composition
+        DISABLE TRIGGER versioning_trigger;
+    ALTER TABLE ehr.entry
+        DISABLE TRIGGER versioning_trigger;
+    ALTER TABLE ehr.event_context
+        DISABLE TRIGGER versioning_trigger;
+    ALTER TABLE ehr.participation
+        DISABLE TRIGGER versioning_trigger;
+    ALTER TABLE ehr.status
+        DISABLE TRIGGER versioning_trigger;
+
+    RETURN QUERY WITH
+                     -- Query IDs
+                     select_composition_ids
+                         AS (SELECT id FROM ehr.composition WHERE ehr_id = ehr_id_param),
+                     select_contribution_ids
+                         AS (SELECT id FROM ehr.contribution WHERE ehr_id = ehr_id_param),
+
+                     -- Delete data
+
+                     -- ON DELETE CASCADE:
```

Comment:
The comments need to be updated. Would be helpful to mention that, as a prerequisite, no folders must exist.